lib/everest.cpp

```diff
@@ -744,6 +746,32 @@ UnsubscribeToken Everest::provide_external_mqtt_handler(const std::string& topic
     return [this, topic, token]() { this->mqtt_abstraction.unregister_handler(topic, token); };
 }
 
+UnsubscribeToken Everest::provide_external_mqtt_handler(const std::string& topic, const StringPairHandler& handler) {
+    BOOST_LOG_FUNCTION();
+
+    // check if external mqtt is enabled
+    if (!this->module_manifest.contains("enable_external_mqtt") &&
+        this->module_manifest["enable_external_mqtt"] == false) {
+        EVLOG_AND_THROW(EverestApiError(fmt::format("Module {} tries to provide an external MQTT handler, but didn't "
+                                                    "set 'enable_external_mqtt' to 'true' in its manifest",
+                                                    this->config.printable_identifier(this->module_id))));
+    }
+
+    std::string external_topic = fmt::format("{}{}", this->mqtt_external_prefix, topic);
+
+    // must be json and not std::string
+    Handler external_handler = [handler](const std::string& topic, const json& data) {
+        EVLOG_verbose << fmt::format("Incoming external mqtt data for topic '{}'...", topic);
+        std::string data_s = (data.is_string()) ? std::string(data) : data.dump();
+        handler(topic, data_s);
+    };
+
+    std::shared_ptr<TypedHandler> token =
+        std::make_shared<TypedHandler>(HandlerType::ExternalMQTT, std::make_shared<Handler>(external_handler));
+    this->mqtt_abstraction.register_handler(external_topic, token, QOS::QOS0);
+    return [this, topic, token]() { this->mqtt_abstraction.unregister_handler(topic, token); };
+}
+
```

Comment:
This is almost 100% copy and paste of the above function.  This could be generalized - probably by using a template.

Reply:
I agree